### ISSUE TITLE
Better debugging tools for viewing state of Deferreds

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4304]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)| @jml reported|
|-|-|
|Trac ID|trac#4304|
|Type|enhancement|
|Created|2010-02-23 18:38:21Z|
|Branch|https://github.com/twisted/twisted/tree/deferred-debug-4304-4|

Deferreds can be quite tricky to debug. Twisted should provide tools for programmers that inspect the state of a Deferred and report it in a meaningful way.

Although graphical output would be nice, we can consider this ticket closed when we have a function that dumps some text for a Deferred that includes:
  * whether the Deferred is paused
  * whether the Deferred is running
  * the callback chain
  * the current value
  * whether the Deferred is waiting on another

<details><summary>Searchable metadata</summary>

```
trac-id__4304 4304
type__enhancement enhancement
reporter__jml jml
priority__normal normal
milestone__ 
branch__deferred_debug_4304_4 deferred-debug-4304-4
branch_author__radix__therve__jml radix, therve, jml
status__new new
resolution__None None
component__core core
keywords__ 
time__1266950301000000 1266950301000000
changetime__1380262367000000 1380262367000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="jml's avatar" src="https://avatars.githubusercontent.com/u/125674?s=50" width="50" height="50">](https://github.com/jml)<a name="note_1"></a>|@jml commented|
|-|-|

(In [28468]) Branching to 'deferred-debug-4304'

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_3"></a>|@radix commented|
|-|-|

(In [32842]) Branching to 'deferred-debug-4304-2'

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_4"></a>|@radix commented|
|-|-|

(In [36286]) Branching to 'deferred-debug-4304-3'

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_5"></a>|@therve commented|
|-|-|

(In [38516]) Branching to 'deferred-debug-4304-4'

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)<a name="note_6"></a>|@radix commented|
|-|-|

I created a new ticket to track one part of the implementation: bug [#6538](https://github.com/twisted/twisted/issues/6538).
